src/Symfony/Component/Translation/Bridge/Phrase/Event/AbstractPhraseEvent.php

```diff
@@ -0,0 +1,36 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Translation\Bridge\Phrase\Event;
+
+use Symfony\Component\Translation\TranslatorBag;
+use Symfony\Contracts\EventDispatcher\Event;
+
+/**
+ * @author wicliff <wicliff.wolda@gmail.com>
+ */
+abstract class AbstractPhraseEvent extends Event
+{
+    public function __construct(
+        private TranslatorBag $bag
+    ) {
+    }
+
+    public function getBag(): TranslatorBag
+    {
+        return $this->bag;
+    }
+
+    public function setBag(TranslatorBag $bag): void
+    {
+        $this->bag = $bag;
+    }
```

Comment:
not sure if the setter is needed as the use case is mostly to mutate the bag not replace it, or?

mutation would be possible with the getter already

Reply:
yes agreed. mutation was the use case i was thinking of and added the setter for flexibility, but that can be re-visited once a valid use case presents itself